src/components/EditPasswordForm/EditPasswordForm.tsx

```diff
@@ -27,13 +29,32 @@ type Props = {
 };
 
 const EditPasswordForm: React.FC<Props> = ({ onSubmit, onChange, onBlur, showOldPasswordField, showResetTokenField, value, errors, submitting }: Props) => {
-  const { t } = useTranslation('account');
+  const { t } = useTranslation(['account', 'user']);
+  const email = useAccountStore((state) => state.user?.email);
+  const [errorMessage, setErrorMessage] = useState<string>();
+  const feedbackMessage = errors.form || errorMessage;
+  const resendEmailClickHandler = async () => {
+    try {
+      if (email) {
+        await resetPassword(email, '');
+      }
+    } catch (error: unknown) {
+      setErrorMessage(t('user:account.resend_mail_error'));
+    }
+  };
   return (
     <form onSubmit={onSubmit} data-testid={testId('forgot-password-form')} noValidate className={styles.forgotPasswordForm}>
-      <h2 className={styles.title}>{t('reset.password_reset')}</h2>
-      {errors.form ? <FormFeedback variant="error">{errors.form}</FormFeedback> : null}
-
-      {showOldPasswordField ? (
+      {feedbackMessage && <FormFeedback variant="error">{feedbackMessage}</FormFeedback>}
+      <h2 className={styles.title}>{!showOldPasswordField || !showResetTokenField ? t('reset.password_reset') : t('user:account.add_password')}</h2>
```

Comment:
This condition seems a bit off to me. Shouldn't we make it explicit (via props) that a password will be added instead of edited? I'm also not against having a separate `<AddPasswordForm />` component if that will make it easier and will not be mostly duplicate code.